009-ui-and-admin-server-security.md

```diff
@@ -0,0 +1,107 @@
+# Security Extension
+
+This proposal is supplementary to the security documentation in the [Using Strimzi](https://strimzi.io/docs/operators/latest/using.html#security-str) guide.
+
+## Problem description
+This proposal discusses amendments to support authentication and authorization with the new [Strimzi UI](https://github.com/strimzi/proposals/pull/6) and [Strimzi Admin Server](https://github.com/strimzi/proposals/pull/9).
+
+### Authentication
+Strimzi currently supports SCRAM-SHA-512, OAuth2 and Mutual TLS for authentication of clients connecting to the Kafka brokers.
+The Strimzi cluster operator can be configured to support all 3 mechanisms for authentication for clients inside or outside the Kubernetes cluster.
+The SCRAM-SHA-512 and Mutual TLS support uses the native support in Kafka for authentication and for OAuth2, Strimzi provides an [OAuth2 library](https://github.com/strimzi/strimzi-kafka-oauth) which has a generic OAuth2 interface allowing you to connect to multiple different OAuth2 servers. 
+Browser based UIs do not traditionally use mutual TLS so the proposed mechanisms for authentication on the UI only covers OAuth2 and SCRAM-SHA-512.
+The admin server proposal similarly only covers SCRAM-SHA-512 and OAuth2 support.
+The OAuth2 and SCRAM-SHA-512 mechanisms for the admin server propagates the SCRAM credentials or the OAuth token to Kafka where the Kafka broker performs the authentication.
+Implementing mutual TLS support in the admin server would require a mechanism that could propagate the identity of the client to Kafka as the admin server will be the TLS termination point and authentication would be between the external client and the admin server. Implementing this in a secure way is not trivial and is not included in this proposal.
+
+Both the UI and the admin server will also support a scenario with no security.
+
+### Authorization
+Both security mechanisms allow authorization to be done using the standard Kafka authorizers along with ACL support.
+Alternatives are available with Kafka authorizers being supplied for [KeyCloak](https://www.keycloak.org/) and [Open Policy Agent](https://www.openpolicyagent.org/docs/latest/) as documented in the Strimzi documentation. The proposed authentication mechanisms for the UI and the admin server will allow these documented authorization methods to be used without requiring any additional changes to the documented methods.
+
+## Proposed Authentication Implementations
+### Admin server using SCRAM-SHA-512
+The following describes the flow for authentication and authorization using SCRAM SHA512 as the authentication type when a web client calls a Kafka client API through the admin server.
+In the diagram:
+1. The credentials are encoded in a base64 encoded Basic Auth header and are passed to the admin server.
```

Comment:
I would assume that the connection between web client and admin server is TLS encrypted right? Wouldn't it be better to make it clear here?

Reply:
I think the answer to that is a community decision but I agree that it should be TLS encrypted as that is a best practise. I guess the one scenario we may not do that is in a "security off" scenario. If the general agreement is that it should be encrypted for all scenarios except the "security off" then I will add specific words to that affect in the proposal.

Reply:
I would say that for admin features we should not consider the "security off" scenario. If we are going to send credentials just base64 encoded we should be sure to have TLS imho.

Reply:
@ppatierno I added a statement that the connection is encrypted. I also tried to clarify what I meant by `security off` so hopefully this will satisfy your request.

@mstruk `Basic Auth` was a shortened version of `Basic Authentication` which is the schema on the `Authorization` header. I changed the sentence to be more explicit.